src/Interpreters/Context.h

```diff
@@ -1336,7 +1336,14 @@ class Context: public ContextData, public std::enable_shared_from_this<Context>
     std::shared_ptr<Clusters> getClustersImpl(std::lock_guard<std::mutex> & lock) const;
 
     /// Throttling
+
+    size_t step_count = 0;
```

Comment:
let's not put it inside `Context`. e.g. it could be a static data member of `IQueryPlanStep`. the same for `IProcessor`

Reply:
Put in context can make explain result more stable, every time we explain same query will get same id.
If we put in static data, we can not get stable result, so I put it in context.

Reply:
I think we better have unstable results than polluting context with this random counters

Reply:
For stateless case `01786_explain_merge_tree`, if not stable, the result may not have fixed result. So How to fix this case? just disable json output?

Reply:
> just disable json output

I guess it makes no difference to the test logic what output format we use

Reply:
Just worries background thread call Plan. I will remove json output.

Reply:
> I think we better have unstable results than polluting context with this random counters

Recently I think stable result is a important feature, If we worry about polluting context, how about put a point of int in CurrentThread::ThreadStatus. This should not polluting context and get a better stable index.

Reply:
> Recently I think stable result is a important feature

what value exactly do you see in it?

Reply:
It's easy to Debugging.
When I found some processors slow, we can use explain query to identify its' steps. if we use static value , It hard to do it if query is complex.

If you think it is not a important, I will change it to static value.

Reply:
I put counter into CurrentThread instead of Context. This may help for this index.

Reply:
> It's easy to Debugging.
When I found some processors slow, we can use explain query to identify its' steps. if we use static value , It hard to do it if query is complex.

yep, i think you're absolutely right